prow/cmd/phaino/local.go

```diff
@@ -246,23 +248,108 @@ func convertToLocal(ctx context.Context, log *logrus.Entry, pj prowapi.ProwJob,
 		log.Warn("Ignoring resource requirements")
 	}
 
+	// A map of volume mounts for the run.
+	// Key is the mount path and value is the local path.
+	volumeMounts := make(map[string]string)
 	for _, mount := range container.VolumeMounts {
+		if strExists(opt.skippedVolumesMounts, mount.Name) {
+			logrus.Infof("Volume mount %q skipped", mount.Name)
+			continue
+		}
 		vol := volume(*pj.Spec.PodSpec, mount.Name)
 		if vol == nil {
 			return nil, fmt.Errorf("mount %q missing associated volume", mount.Name)
 		}
 		if vol.EmptyDir != nil {
-			localArgs = append(localArgs, "-v", mount.MountPath)
+			volumeMounts[mount.MountPath] = ""
 		} else {
 			local, err := readMount(ctx, mount)
 			if err != nil {
 				return nil, fmt.Errorf("bad mount %q: %v", mount.Name, err)
 			}
-			arg := local + ":" + mount.MountPath
+			mountPath := mount.MountPath
 			if mount.ReadOnly {
-				arg += ":ro"
+				mountPath += ":ro"
+			}
+			volumeMounts[mountPath] = local
+		}
+	}
+	for pathInContainer, localPath := range opt.extraVolumesMounts {
+		volumeMounts[pathInContainer] = localPath
+	}
+
+	// A map of env vars for the run.
+	// Key is the env name and value is the env value.
+	envs := make(map[string]string)
+	for _, env := range container.Env {
+		if strExists(opt.skippedEnvVars, env.Name) {
+			continue
+		}
+		envs[env.Name] = env.Value
+	}
+	for name, value := range opt.extraEnvVars {
+		envs[name] = value
+	}
+	// Overwrite GOPATH in the container.
+	envs["GOPATH"] = opt.gopath
+
+	// Setup gcloud credentials
+	if opt.useLocalGcloudCredentials {
+		gcloudKey := os.Getenv(gcloudCredEnvName)
+		// If GOOGLE_APPLICATION_CREDENTIALS is not empty, also mount the key file
+		// to the container
+		if gcloudKey != "" {
+			if _, err := os.Stat(gcloudKey); !os.IsNotExist(err) {
+				volumeMounts[gcloudKey+":ro"] = gcloudKey
+				envs[gcloudCredEnvName] = gcloudKey
+			} else {
+				logrus.Warningf("The GOOGLE_APPLICATION_CREDENTIALS file does not exist on your local machine, thus gcloud authentication won't work in the container")
```

Comment:
idea: How does `gcloud auth application-default login` work? Might be nice to enable this to work eventually (I'll sometimes use this with testgrid development, simulates workload identity)